bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -48,41 +43,28 @@
     private final CodecRegistry registry;
     private final PropertyCodecRegistry propertyCodecRegistry;
     private final DiscriminatorLookup discriminatorLookup;
-    private final ConcurrentMap<ClassModel<?>, Codec<?>> codecCache;
     private final boolean specialized;
 
     PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
                   final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
         this.classModel = classModel;
-        this.registry = fromRegistries(fromCodecs(this), codecRegistry);
+        this.registry = codecRegistry;
         this.discriminatorLookup = discriminatorLookup;
-        this.codecCache = new ConcurrentHashMap<ClassModel<?>, Codec<?>>();
         this.propertyCodecRegistry = new PropertyCodecRegistryImpl(this, registry, propertyCodecProviders);
         this.specialized = shouldSpecialize(classModel);
         specialize();
     }
 
     PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry registry, final PropertyCodecRegistry propertyCodecRegistry,
-                  final DiscriminatorLookup discriminatorLookup, final ConcurrentMap<ClassModel<?>, Codec<?>> codecCache,
-                  final boolean specialized) {
+                  final DiscriminatorLookup discriminatorLookup, final boolean specialized) {
         this.classModel = classModel;
-        this.registry = fromRegistries(fromCodecs(this), registry);
+        this.registry = fromRegistries(registry, fromCodecs(this));
```

Comment:
Why is the order swapped here?

Reply:
Updated - no need to nest the registry - as per the other constructor.